enginetest/queries/script_queries.go

```diff
@@ -3392,6 +3392,141 @@ var CreateCheckConstraintsScripts = []ScriptTest{
 	},
 }
 
+var PreparedScriptTests = []ScriptTest{
+	{
+		Name:        "simple select case no bindings",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:       "execute s",
+				ExpectedErr: sql.ErrUnknownPreparedStatement,
+			},
+			{
+				Query: "prepare s from 'select 1'",
+				Expected: []sql.Row{
+					{sql.OkResult{Info: plan.PrepareInfo{}}},
+				},
+			},
+			{
+				Query: "execute s",
+				Expected: []sql.Row{
+					{1},
+				},
+			},
+			{
+				Query: "deallocate prepare s",
+				Expected: []sql.Row{
+					{sql.OkResult{}},
+				},
+			},
+			{
+				Query:       "execute s",
+				ExpectedErr: sql.ErrUnknownPreparedStatement,
+			},
+		},
+	},
+	{
+		Name: "simple select case one binding",
+		SetUpScript: []string{
+			"set @a = 1",
+			"set @b = 100",
+			"set @c = 'abc'",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "prepare s from 'select ?'",
+				Expected: []sql.Row{
+					{sql.OkResult{Info: plan.PrepareInfo{}}},
+				},
+			},
+			{
+				Query:       "execute s",
+				ExpectedErr: sql.ErrInvalidArgument,
+			},
+			{
+				Query: "execute s using @abc",
+				Expected: []sql.Row{
+					{nil},
+				},
+			},
+			{
+				Query:       "execute s using @a, @b, @c, @abc",
+				ExpectedErr: sql.ErrInvalidArgument,
+			},
+			{
+				Query: "execute s using @a",
+				Expected: []sql.Row{
+					{1},
+				},
+			},
+			{
+				Query: "execute s using @b",
+				Expected: []sql.Row{
+					{100},
+				},
+			},
+			{
+				Query: "execute s using @c",
+				Expected: []sql.Row{
+					{"abc"},
+				},
+			},
+			{
+				Query: "deallocate prepare s",
+				Expected: []sql.Row{
+					{sql.OkResult{}},
+				},
+			},
+			{
+				Query:       "execute s using @a",
+				ExpectedErr: sql.ErrUnknownPreparedStatement,
+			},
+		},
+	},
+	{
+		Name: "prepare insert",
```

Comment:
Would be nice to add a more complex test, with some subqueries and maybe a derived table or something in it, just to test something a little more complex.

Reply:
I see some error case tests for `execute`; it would be good to add some for `prepare` with an invalid statement and maybe a quick one for `deallocate` with an unknown statement id.